src/js/plugins/plugin.vendor-fullscreen.js

```diff
@@ -0,0 +1,231 @@
+/* global BookReader */
+
+/**
+ * Toggles browser's native fullscreen mode if available
+ */
+
+const EVENT_NAMESPACE = '.bookreader_vendor-fullscreen';
+// function isMobile() {
+//   return (typeof window.orientation !== 'undefined') || (navigator.userAgent.indexOf('IEMobile') !== -1);
+// };
+
+// if (this.isMobile()) { return; }
+
+jQuery.extend(BookReader.defaultOptions, {
+  /** @type {true} */
+  enableVendorFullscreenPlugin: true
+});
+
+/** @override */
+BookReader.prototype.setup = (function(super_) {
+  return function(options) {
+    super_.call(this, options);
+
+    this.isVendorFullscreenActive = false;
+  };
+})(BookReader.prototype.setup);
+
+/** @override */
+BookReader.prototype.getInitialMode = (function(super_) {
+  return function(params) {
+    let nextMode = super_.call(this, params);
+    if (this.isVendorFullscreenActive) {
+      nextMode = this.constMode1up;
+    }
+    return nextMode;
+  };
+})(BookReader.prototype.getInitialMode);
+
+/** @override */
+BookReader.prototype.init = (function(super_) {
+  return function() {
+    super_.call(this);
+
+    if (!BookReader.util.fullscreenAllowed()) {
+      return;
+    }
+    // In fullscreen mode the colorbox and overlay need to be inside the fullscreen element to display properly.
+    BookReader.util.bindFullscreenChangeListener(this, (e) => {
+      e.data.resize();
+      e.data.updateBrClasses();
+      const cboxOverlay = $('#cboxOverlay');
+      const cbox = $('#colorbox');
+      if (BookReader.util.isFullscreenActive()) {
+        // In full screen mode, the colorbox and overlay need
+        // to be children of the fullscreen element to display properly.
+        let $fullscreen = $(BookReader.util.getFullscreenElement());
+        $fullscreen.append(cboxOverlay).append(cbox);
+      } else {
+        // In non-fullscreen mode, the colorbox and overlay need
+        // to be children of the main document body.
+        $(document.body).append(cboxOverlay).append(cbox);
+      }
+    });
+  }
+})(BookReader.prototype.init);
+
+/** 
+ * Start fullscreen mode
+ */
+BookReader.prototype.enterFullWindow = function() {
+  this.refs.$brContainer.css('opacity', 0);
+
+  const windowWidth = $(window).width();
+  if (windowWidth <= this.onePageMinBreakpoint) {
+    this.switchMode(this.constMode1up);
+  }
+
+  this.isVendorFullscreenActive = true;
+  this.updateBrClasses();
+
+  this.resize();
+  this.jumpToIndex(this.currentIndex());
+
+  this.refs.$brContainer.animate({ opacity: 1 }, 400, 'linear');
+
+  this._fullWindowExitHandler = function (e) {
+    if (e.keyCode === 27) this.exitFullScreen();
+  };
+  $(document).on('keyup.' + EVENT_NAMESPACE, this._fullWindowExitHandler);
```

Comment:
Let's not attach this to BR:

```suggestion
  $(document).on(`keyup.${EVENT_NAMESPACE}`, e => {
        if (e.keyCode === 27) this.exitFullScreen();
  });
```